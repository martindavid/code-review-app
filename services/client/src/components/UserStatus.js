import React from 'react';
import axios from 'axios';
import { Link } from 'react-router-dom';

class UserStatus extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      email: '',
      id: '',
      username: '',
      active: '',
      admin: ''
    }
  };

  componentDidMount() {
    if (this.props.isAuthenticated) {
      this.getUserStatus();
    }
  }

  getUserStatus = (e) => {
    const options = {
      url: `${process.env.REACT_APP_USERS_SERVICE_URL}/auth/status`,
      method: 'get',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${window.localStorage.authToken}`
      }
    }

    return axios(options)
      .then((res) => this.setState({
        email: res.data.data.email,
        id: res.data.data.id,
        username: res.data.data.username,
        admin: String(res.data.data.admin),
        active: String(res.data.data.active)
      }))
      .catch((err) => console.log(err))
  }

  render() {
    const { email, id, username, active, admin } = this.state;
    if (!this.props.isAuthenticated) {
      return (
        <p>You must be logged in to view this. Click <Link to='/login'>here</Link> to log back in</p>
      );
    }
    return (
      <div>
        <ul>
          <li><strong>User ID: </strong>{id}</li>
          <li><strong>Email: </strong>{email}</li>
          <li><strong>Username: </strong>{username}</li>
          <li><strong>Active: </strong>{active}</li>
          <li><strong>Admin: </strong>{admin}</li>
        </ul>
      </div>
    );
  }
}

export default UserStatus;
