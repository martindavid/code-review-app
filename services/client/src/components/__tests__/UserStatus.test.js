import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import UserStatus from '../UserStatus';


describe('When authenticated', () => {
  test('UserStatus render properly', () => {
    const component = (
        <UserStatus isAuthenticated={true} />
    );
    const wrapper = shallow(component);
    const element = wrapper.find('li');
    expect(element.length).toBe(3);
    expect(element.get(0).props.children[0].props.children).toBe('User ID: ');
  });

  test('UserStatus render a snapshot properly', () => {
    const component = (
      <Router>
        <UserStatus isAuthenticated={true} />
      </Router>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('When not authenticated', () => {
  test('UserStatus render properly', () => {
    const wrapper = shallow(<UserStatus isAuthenticated={false} />);
    const element = wrapper.find('p');
    expect(element.length).toBe(1);
  expect(element.get(0).props.children[0]).toContain('You must be logged in to view this.');
  });
  
  test('UserStatus render a snapshot properly', () => {
    const component = (
      <Router>
        <UserStatus isAuthenticated={false} />
      </Router>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
