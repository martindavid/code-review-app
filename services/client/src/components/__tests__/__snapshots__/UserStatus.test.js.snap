// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When authenticated UserStatus render a snapshot properly 1`] = `
<div>
  <ul>
    <li>
      <strong>
        User ID: 
      </strong>
      
    </li>
    <li>
      <strong>
        Email: 
      </strong>
      
    </li>
    <li>
      <strong>
        Username: 
      </strong>
      
    </li>
    <li>
      <strong>
        Active: 
      </strong>
      
    </li>
    <li>
      <strong>
        Admin: 
      </strong>
      
    </li>
  </ul>
</div>
`;

exports[`When not authenticated UserStatus render a snapshot properly 1`] = `
<p>
  You must be logged in to view this. Click 
  <a
    href="/login"
    onClick={[Function]}
  >
    here
  </a>
   to log back in
</p>
`;
